src/wallet/coinselection.cpp

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2012-2016 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "wallet/coinselection.h"
+#include "util.h"
+#include "utilmoneystr.h"
+
+// Descending order comparator
+struct {
+    bool operator()(CInputCoin a, CInputCoin b) const
```

Comment:
Here and throughout the PR, probably would be better to pass CInputCoin by const reference to avoid copying the struct. Probably want `for (const CInputCoin& utxo : utxo_pool)` instead of `for (CInputCoin utxo : utxo_pool)` as well.

Reply:
Done.